src/components/date-range/date-range.test.tsx

```diff
@@ -0,0 +1,1125 @@
+  const ref = { current: null };
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startRef={ref}
+    />
+  );
+
+  expect(ref.current).toBe(screen.getByRole("textbox", { name: "start" }));
+});
+
+test("should support ref as a callback on the 'start' input", () => {
+  const ref = jest.fn();
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startRef={ref}
+    />
+  );
+
+  expect(ref).toHaveBeenCalledWith(
+    screen.getByRole("textbox", { name: "start" })
+  );
+});
+
+test("should set the ref on the 'start' input to null after the component is unmounted", () => {
+  const ref = { current: null };
+  const { unmount } = render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      startRef={ref}
+    />
+  );
+  unmount();
+
+  expect(ref.current).toBeNull();
+});
+
+test("should support ref as an object on the 'end' input", () => {
+  const ref = { current: null };
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      endRef={ref}
+    />
+  );
+
+  expect(ref.current).toBe(screen.getByRole("textbox", { name: "end" }));
+});
+
+test("should support ref as a callback on the 'end' input", () => {
+  const ref = jest.fn();
+  render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      endRef={ref}
+    />
+  );
+
+  expect(ref).toHaveBeenCalledWith(
+    screen.getByRole("textbox", { name: "end" })
+  );
+});
+
+test("should set the ref on the 'end' input to null after the component is unmounted", () => {
+  const ref = { current: null };
+  const { unmount } = render(
+    <DateRange
+      startLabel="start"
+      onChange={() => {}}
+      endLabel="end"
+      value={["10/10/2016", "11/11/2016"]}
+      endRef={ref}
+    />
+  );
+  unmount();
+
+  expect(ref.current).toBeNull();
+});
+
+test("should only display the start input tooltip when the user hovers over it and both inputs have error strings passed with `validationRedesignOptIn` not set", async () => {
+  const user = userEvent.setup();
+  render(
+    <CarbonProvider>
+      <DateRange
+        startLabel="start"
+        onChange={() => {}}
+        endLabel="end"
+        value={["10/10/2016", "11/11/2016"]}
+        startError="start error"
+        endError="end error"
+      />
+    </CarbonProvider>
+  );
+
+  const start = screen.getByRole("textbox", { name: "start" });
+  await user.hover(start);
+  const tooltips = screen.getAllByRole("tooltip");
+
+  expect(tooltips).toHaveLength(1);
+  expect(tooltips[0]).toBeVisible();
+  expect(tooltips[0]).toHaveTextContent("start error");
```

Comment:
**suggestion(non-breaking)**: what do you think about querying the tooltip directly? This would help improve readability 

```suggestion
  const tooltip = screen.getByRole("tooltip", { name: "start error" });

  expect(tooltip).toBeVisible();
```

Reply:
Yep good point not sure why I've deviated from this approach here tbh